### ISSUE TITLE
max-asserts incorrectly counts called functions on t.context as assertions

### ISSUE THREAD

**Issue by MethodGrab** [avajs/eslint-plugin-ava#142]
### Problem
`ava/max-asserts` incorrectly counts called functions on `t.context` as assertions.

### Example:
```js
/* eslint "ava/max-asserts" : [ "warn", 5 ] */

import test from 'ava';

test.beforeEach( t => {
	t.context.bar = () => 'bar';
});


// this will be fine
test( 'max-asserts ok', t => {
	const foo = 'foo';

	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
});


// this will count 6 assertions
test( 'max-asserts incorrect count', t => {
	const foo = 'foo';

	// this is counted as an assertion
	const bar1 = t.context.bar();

	// this is not counted as an assertion
	const Bar = t.context.bar;
	const bar2 = Bar();

	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
	t.true( typeof foo === 'string' );
});
```

### Expected
No warnings

### Actual
The following warnings are displayed when linting the above file:
```
37:2  warning  Expected at most 5 assertions, but found 6  ava/max-asserts
✖ 1 problem (0 errors, 1 warning)
```

**Comment by jamestalmage** [comment]
Probably related to the use of `util.nameOfRootObject` [here](https://github.com/avajs/eslint-plugin-ava/blob/master/rules/max-asserts.js#L27).

@jfmengels - any reason you did it that way?

**Comment by jfmengels** [comment]
Ah right!
Yeah, at the time I didn't know about t.context. I think we probably have some utils (in use-t-well?) that check whether the call is made on t.context members.

**Comment by sindresorhus** [comment]
@jfmengels I think it would be more future proof to do a whitelist. I would extract [this](https://github.com/avajs/eslint-plugin-ava/blob/a443d7a9c94165f42749938e6b491a7c10749b6c/rules/assertion-message.js#L6) into `util.js` and also export a version that is `Object.keys()` of that with name of all supported assertion methods.
